lib/toolbox/name_generator.c

```diff
@@ -0,0 +1,82 @@
+#include "name_generator.h"
+
+#include <stdio.h>
+#include <stdint.h>
+#include <furi_hal_rtc.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <furi.h>
+
+void set_random_name(char* name, uint8_t max_name_size) {
+    const char* prefix[] = {
+        "ancient",     "hollow",     "strange",    "disappeared", "unknown",
+        "unthinkable", "unnameable", "nameless",   "my",          "concealed",
+        "forgotten",   "hidden",     "mysterious", "obscure",     "random",
+        "remote",      "uncharted",  "undefined",  "untraveled",  "untold",
+    };
+
+    const char* suffix[] = {
+        "door",
+        "entrance",
+        "doorway",
+        "entry",
+        "portal",
+        "entree",
+        "opening",
+        "crack",
+        "access",
+        "corridor",
+        "passage",
+        "port",
+    };
+    uint8_t prefix_i = rand() % COUNT_OF(prefix);
+    uint8_t suffix_i = rand() % COUNT_OF(suffix);
+
+    snprintf(name, max_name_size, "%s_%s", prefix[prefix_i], suffix[suffix_i]);
+    // Set first symbol to upper case
+    name[0] = name[0] - 0x20;
+}
+
+const char* convert_app_extension_to_name(char* app_name) {
+    struct NameConversion {
+        const char* original;
+        const char* converted;
+    };
+
+    struct NameConversion conversions[] = {
+        {".ibtn", "iBtn"}, {".nfc", "NFC"}, {".rfid", "RFID"}, {".sub", "RAW"}};
```

Comment:
That doesn't looks extensible. Why not to pass it explicitly from application side?